Upgraded theme in 24.0.2 renders extra newline, has schema validation errors

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### What happened?

I just upgraded to 24.0.2 and now a new line is added to my prompt where a segment's `newline: true` was conditional in 23.20.3 before. The diff between the new theme and .bak file is also significant, and the upgrade filed added an empty string `""` for `foreground` in a couple places, which is a schema violation.

I'm using Windows Terminal on Windows 11 but running OMP 24.0.2 in Ubuntu 22.04.

The log below is during the theme upgrade.

### Theme

<details><summary>theme.omp.yml</summary>

```yaml
# yaml-language-server: $schema=https://raw.githubusercontent.com/JanDeDobbeleer/oh-my-posh/main/themes/schema.json
version: 2
console_title_template: >-
  {{ $pwd := .PWD -}}
  {{ .Shell }} in {{ with $repo := .Segments.Git -}}
    {{ if (gt (sub (len $pwd) (len $repo.Dir)) 20) -}}
      git:{{ $repo.RepoName }}/ΓÇª/{{ base $pwd }}
    {{- else -}}
      git:{{ $repo.RepoName }}{{ trimPrefix $repo.Dir $pwd}}
    {{- end }}
  {{- else -}}
    {{ .Folder }}
  {{- end }}
final_space: true
shell_integration: true

palette:
  # https://colorkit.co/palette/02944e-4f0193-f1c516-eb850d-e44403-05a3e2-f7f6f5-7c7b7b-464646-101010/
  green: "#02944e"
  purple: "#4f0193"
  darkPurple: "#940248" # not in palette
  yellow: "#f1c516"
  orange: "#eb850d"
  red: "#e44403"
  blue: "#05a3e2"
  white: "#f7f6f5"
  gray: "#7c7b7b"
  darkGray: "#464646"
  black: "#101010"

blocks:
- type: prompt
  alignment: left
  segments:
  - type: shell
    style: diamond
    foreground: p:white
    background: p:purple
    properties:
      mapped_shell_names:
        powershell: PS
        pwsh: PS

  - type: python
    style: powerline
    powerline_symbol: "εé░"
    foreground: p:white
    background: p:darkPurple
    template: " εê╡ {{ if .Error }}{{ .Error }}{{ else }}{{ if .Venv }}{{ .Venv }} {{ end }}{{ .Full }}{{ end }} "

  - type: executiontime
    style: powerline
    powerline_symbol: "εé░"
    foreground: p:black
    background: p:orange
    properties:
      style: austin

  - type: status
    style: powerline
    powerline_symbol: "εé░"
    foreground: p:white
    background: p:blue
    background_templates:
    - "{{ if gt .Code 0 }}p:red{{ end }}"
    properties:
      always_enable: true

  - type: session
    style: powerline
    powerline_symbol: "εé░"
    foreground: p:white
    background: p:blue
    template: " {{ if .SSHSession }}∩áù {{ end }}{{ .UserName }} "

  - type: root
    style: powerline
    powerline_symbol: "εé░"
    foreground: p:white
    background: p:red
    template: " ε¡┤ "

  - type: path
    style: powerline
    powerline_symbol: "εé░"
    foreground: p:black
    background: p:gray
    template: ' {{ path .Path .Location }} {{ repeat .StackCount "εé▒" }}'
    properties:
      style: full

- type: prompt
  alignment: left
  newline: true
  segments:
  - type: git
    style: diamond
    trailing_diamond: "εé░"
    foreground: p:white
    foreground_templates:
    - "{{ if or (.Working.Changed) (.Staging.Changed) }}p:black{{ end }}"
    - "{{ if and (gt .Ahead 0) (gt .Behind 0) }}p:white{{ end }}"
    - "{{ if gt .Ahead 0 }}p:black{{ end }}"
    background: p:green
    background_templates:
    - "{{ if or (.Working.Changed) (.Staging.Changed) }}p:yellow{{ end }}"
    - "{{ if and (gt .Ahead 0) (gt .Behind 0) }}p:red{{ end }}"
    - "{{ if gt .Ahead 0 }}p:yellow{{ end }}"
    - "{{ if gt .Behind 0 }}p:orange{{ end }}"
    template: " {{ if .UpstreamURL }}{{ url .UpstreamIcon .UpstreamURL }}{{ end }}{{
      .HEAD }}{{if .BranchStatus }} {{ .BranchStatus }}{{ end }}{{ if .Working.Changed
      }} ∩üä {{ .Working.String }}{{ end }}{{ if .Staging.Changed }} ∩üå {{ .Staging.String
      }}{{ end }} "
    properties:
      branch_icon: "εéá "
      fetch_status: true

debug_prompt:
  background: transparent
  template: "<p:white,p:red> ε¬» DBG </><p:red,transparent>εé░</> "

secondary_prompt:
  background: transparent
  template: "<p:black,p:gray> εé▒</><p:gray,transparent>εé░</> "

# vim: set et sw=2:
```

</details> 

### What OS are you seeing the problem on?

Linux

### Which shell are you using?

bash

### Log output

```Shell
Version: 24.0.2

Shell: bash (5.1.16(1)-release)

Prompt:

]133;D;0]133;A bash εé░ heaths εé░ ]8;;file://wsl.localhost/Ubuntu-22.04/home/heaths/.config/oh-my-posh\~/.config/oh-my-posh]8;;\ εé░
 εéá main Γëí ∩üä ?1 ~1 εé░ ]133;B

Segments:

ConsoleTitle(true)                         -   0 ms
Shell(true)                                -   0 ms
Python(false)                              - 114 ms
Executiontime(false)                       -   0 ms
Status(false)                              -   0 ms
Session(true)                              -   1 ms
Root(false)                                -   0 ms
Path(true)                                 -   0 ms
Git(true)                                  -   2 ms

Run duration: 124.232701ms

Cache path: /home/heaths/.cache/oh-my-posh

Config path: /home/heaths/.config/oh-my-posh/theme.omp.yml

Logs:

[38;2;191;207;240m[DEBUG] [38;2;156;231;201m10:23:27.118 [0mdebug.go:50[38;2;156;231;201m ΓåÆ[0m debug mode enabled
[DEBUG] 10:23:27.118 debug.go:50 ΓåÆ plain mode enabled
[DEBUG] 10:23:27.118 terminal.go:Getenv:161 ΓåÆ NO DATA
[TRACE] 10:23:27.118 terminal.go:Getenv(LOCALAPPDATA) - 13.641┬╡s
[DEBUG] 10:23:27.118 terminal.go:Getenv:161 ΓåÆ NO DATA
[TRACE] 10:23:27.118 terminal.go:Getenv(OMP_CACHE_DIR) - 9.338┬╡s
[DEBUG] 10:23:27.118 terminal.go:Getenv:161 ΓåÆ NO DATA
[TRACE] 10:23:27.118 terminal.go:Getenv(XDG_CACHE_HOME) - 1.657┬╡s
[TRACE] 10:23:27.118 terminal.go:CachePath() - 41.885┬╡s
[DEBUG] 10:23:27.118 terminal.go:65 ΓåÆ loading cache file: /home/heaths/.cache/oh-my-posh/omp.cache
[DEBUG] 10:23:27.118 terminal.go:65 ΓåÆ loading cache key: environment_platform
[DEBUG] 10:23:27.118 terminal.go:65 ΓåÆ loading cache key: is_wsl
[TRACE] 10:23:27.118 terminal.go(/home/heaths/.cache/oh-my-posh/omp.cache) - 86.092┬╡s
[DEBUG] 10:23:27.118 terminal.go:Getenv:161 ΓåÆ NO DATA
[TRACE] 10:23:27.118 terminal.go:Getenv(LOCALAPPDATA) - 2.898┬╡s
[DEBUG] 10:23:27.118 terminal.go:Getenv:161 ΓåÆ NO DATA
[TRACE] 10:23:27.118 terminal.go:Getenv(OMP_CACHE_DIR) - 2.897┬╡s
[DEBUG] 10:23:27.118 terminal.go:Getenv:161 ΓåÆ NO DATA
[TRACE] 10:23:27.118 terminal.go:Getenv(XDG_CACHE_HOME) - 2.594┬╡s
[TRACE] 10:23:27.118 terminal.go:CachePath() - 24.083┬╡s
[DEBUG] 10:23:27.118 terminal.go:65 ΓåÆ loading cache file: /home/heaths/.cache/oh-my-posh/omp.cache.8d3bcd41-0b95-4d58-aaed-22418196bef1
[TRACE] 10:23:27.118 terminal.go(/home/heaths/.cache/oh-my-posh/omp.cache.8d3bcd41-0b95-4d58-aaed-22418196bef1) - 28.329┬╡s
[TRACE] 10:23:27.118 terminal.go:setPromptCount() - 118ns
[TRACE] 10:23:27.118 terminal.go:GOOS() - 38ns
[DEBUG] 10:23:27.118 terminal.go:setPwd:194 ΓåÆ /home/heaths/.config/oh-my-posh
[TRACE] 10:23:27.118 terminal.go:setPwd() - 20.349┬╡s
[DEBUG] 10:23:27.118 terminal.go:Getenv:161 ΓåÆ /home/heaths/.config/oh-my-posh/theme.omp.yml
[TRACE] 10:23:27.118 terminal.go:Getenv(POSH_THEME) - 2.125┬╡s
[DEBUG] 10:23:27.118 terminal_unix.go:Platform:95 ΓåÆ ubuntu
[TRACE] 10:23:27.118 terminal.go:ResolveConfigPath() - 11.136┬╡s
[TRACE] 10:23:27.118 terminal.go:Init() - 286.16┬╡s
[TRACE] 10:23:27.118 terminal.go:Shell() - 131ns
[TRACE] 10:23:27.118 terminal.go:Flags() - 34ns
[TRACE] 10:23:27.122 load.go:loadConfig() - 3.642249ms
[TRACE] 10:23:27.122 terminal.go:Flags() - 57ns
[TRACE] 10:23:27.122 terminal.go:Shell() - 287ns
[DEBUG] 10:23:27.122 debug.go:60 ΓåÆ terminal program: Windows Terminal
[DEBUG] 10:23:27.122 debug.go:60 ΓåÆ terminal shell: shell
[DEBUG] 10:23:27.122 terminal.go:Getenv:161 ΓåÆ NO DATA
[TRACE] 10:23:27.122 terminal.go:Getenv(OMP_CACHE_DISABLED) - 2.441┬╡s
[TRACE] 10:23:27.122 colors.go:SetAccentColor() - 965ns
[TRACE] 10:23:27.122 terminal.go:Shell() - 25ns
[DEBUG] 10:23:27.122 terminal.go:Getenv:161 ΓåÆ 5.1.16(1)-release
[TRACE] 10:23:27.122 terminal.go:Getenv(POSH_SHELL_VERSION) - 2.007┬╡s
[DEBUG] 10:23:27.122 debug.go:PrintDebug:23 ΓåÆ segment: Title
[TRACE] 10:23:27.123 terminal_unix.go:Root() - 654ns
[TRACE] 10:23:27.123 terminal.go:Shell() - 34ns
[TRACE] 10:23:27.123 terminal.go:StatusCodes() - 45ns
[DEBUG] 10:23:27.123 terminal_unix.go:IsWsl:35 ΓåÆ true
[TRACE] 10:23:27.123 terminal_unix.go:IsWsl() - 3.49┬╡s
[DEBUG] 10:23:27.123 terminal_unix.go:IsWsl:35 ΓåÆ true
[TRACE] 10:23:27.123 terminal_unix.go:IsWsl() - 2.172┬╡s
[DEBUG] 10:23:27.124 terminal.go:RunCommand:371 ΓåÆ //wsl.localhost/Ubuntu-22.04/home/heaths/.config/oh-my-posh
[TRACE] 10:23:27.124 terminal.go:RunCommand(wslpath -m /home/heaths/.config/oh-my-posh) - 1.035478ms
[TRACE] 10:23:27.124 terminal.go:GOOS() - 75ns
[TRACE] 10:23:27.124 terminal.go:GOOS() - 51ns
[TRACE] 10:23:27.124 terminal.go:GOOS() - 36ns
[TRACE] 10:23:27.124 terminal.go:GOOS() - 41ns
[TRACE] 10:23:27.124 terminal.go:GOOS() - 40ns
[TRACE] 10:23:27.124 terminal.go:GOOS() - 21ns
[TRACE] 10:23:27.124 terminal.go:GOOS() - 39ns
[TRACE] 10:23:27.124 terminal.go:GOOS() - 36ns
[TRACE] 10:23:27.124 terminal.go:GOOS() - 44ns
[TRACE] 10:23:27.124 terminal.go:GOOS() - 41ns
[TRACE] 10:23:27.124 terminal.go:GOOS() - 24ns
[TRACE] 10:23:27.124 terminal.go:GOOS() - 32ns
[DEBUG] 10:23:27.124 terminal.go:User:331 ΓåÆ heaths
[TRACE] 10:23:27.124 terminal.go:User() - 5.211┬╡s
[DEBUG] 10:23:27.124 terminal.go:Host:348 ΓåÆ heathsl4
[TRACE] 10:23:27.124 terminal.go:Host() - 6.735┬╡s
[TRACE] 10:23:27.124 terminal.go:GOOS() - 63ns
[DEBUG] 10:23:27.124 terminal_unix.go:Platform:95 ΓåÆ ubuntu
[DEBUG] 10:23:27.124 terminal.go:Getenv:161 ΓåÆ 0
[TRACE] 10:23:27.124 terminal.go:Getenv(SHLVL) - 6.24┬╡s
[TRACE] 10:23:27.124 terminal.go:TemplateCache() - 1.163543ms
[TRACE] 10:23:27.124 text.go:Render({{ $pwd := .PWD -}} {{ .Shell }} in {{ with $repo := .Segments.Git -}}   {{ if (gt (sub (len $pwd) (len $repo.Dir)) 20) -}}     git:{{ $repo.RepoName }}/ΓÇª/{{ base $pwd }}   {{- else -}}     git:{{ $repo.RepoName }}{{ trimPrefix $repo.Dir $pwd}}   {{- end }} {{- else -}}   {{ .Folder }} {{- end }}) - 1.479918ms
[TRACE] 10:23:27.124 terminal.go:Flags() - 95ns
[TRACE] 10:23:27.124 terminal.go:StatusCodes() - 56ns
[TRACE] 10:23:27.124 terminal.go:Flags() - 38ns
[DEBUG] 10:23:27.124 terminal.go:Getenv:161 ΓåÆ NO DATA
[TRACE] 10:23:27.124 terminal.go:Getenv(POSH_CURSOR_LINE) - 4.002┬╡s
[DEBUG] 10:23:27.124 terminal.go:Getenv:161 ΓåÆ NO DATA
[TRACE] 10:23:27.124 terminal.go:Getenv(POSH_CURSOR_COLUMN) - 2.824┬╡s
[TRACE] 10:23:27.124 terminal.go:Flags() - 40ns
[TRACE] 10:23:27.124 terminal.go:Flags() - 39ns
[TRACE] 10:23:27.124 terminal.go:Flags() - 85ns
[TRACE] 10:23:27.124 terminal.go:GOOS() - 90ns
[TRACE] 10:23:27.124 terminal.go:Flags() - 203ns
[DEBUG] 10:23:27.124 segment.go:Execute:103 ΓåÆ segment: Path
[TRACE] 10:23:27.124 terminal.go:GOOS() - 115ns
[DEBUG] 10:23:27.124 segment.go:Execute:103 ΓåÆ segment: Executiontime
[DEBUG] 10:23:27.124 path.go:145 ΓåÆ display_cygpath: false
[TRACE] 10:23:27.124 terminal.go:GOOS() - 51ns
[DEBUG] 10:23:27.124 executiontime.go:Enabled:55 ΓåÆ always_enabled: false
[TRACE] 10:23:27.124 terminal.go:GOOS() - 136ns
[TRACE] 10:23:27.124 terminal.go:ExecutionTime() - 93ns
[TRACE] 10:23:27.124 terminal.go:Flags() - 179ns
[TRACE] 10:23:27.124 terminal.go:PathSeparator() - 4.932┬╡s
[DEBUG] 10:23:27.124 executiontime.go:Enabled:57 ΓåÆ threshold: 500.000000
[TRACE] 10:23:27.124 terminal.go:Shell() - 184ns
[TRACE] 10:23:27.124 terminal.go:GOOS() - 70ns
[TRACE] 10:23:27.124 terminal.go:Shell() - 45ns
[TRACE] 10:23:27.124 terminal.go:Flags() - 51ns
[TRACE] 10:23:27.124 terminal.go:GOOS() - 32ns
[DEBUG] 10:23:27.124 segment.go:Execute:103 ΓåÆ segment: Session
[DEBUG] 10:23:27.124 path.go:setMappedLocations:552 ΓåÆ mapped_locations_enabled: true
[TRACE] 10:23:27.124 terminal.go:GOOS() - 43ns
[DEBUG] 10:23:27.124 segment.go:Execute:103 ΓåÆ segment: Shell
[DEBUG] 10:23:27.124 path.go:setMappedLocations:553 ΓåÆ ∩Çô
[DEBUG] 10:23:27.124 terminal.go:Getenv:161 ΓåÆ NO DATA
[TRACE] 10:23:27.124 terminal.go:Getenv(SSH_CONNECTION) - 3.605┬╡s
[DEBUG] 10:23:27.124 terminal.go:Getenv:161 ΓåÆ NO DATA
[TRACE] 10:23:27.124 terminal.go:Getenv(SSH_CLIENT) - 2.909┬╡s
[DEBUG] 10:23:27.124 terminal_unix.go:Platform:95 ΓåÆ ubuntu
[TRACE] 10:23:27.124 terminal.go:GOOS() - 39ns
[TRACE] 10:23:27.124 terminal.go:PathSeparator() - 17.387┬╡s
[TRACE] 10:23:27.124 terminal.go:GOOS() - 29ns
[DEBUG] 10:23:27.124 shell.go:Enabled:26 ΓåÆ mapped_shell_names: map[powershell:PS pwsh:PS]
[TRACE] 10:23:27.124 terminal.go:Shell() - 115ns
[TRACE] 10:23:27.124 terminal.go:Flags() - 98ns
[TRACE] 10:23:27.124 terminal.go:GOOS() - 56ns
[TRACE] 10:23:27.124 terminal.go:TemplateCache() - 138ns
[TRACE] 10:23:27.124 terminal.go:GOOS() - 32ns
[DEBUG] 10:23:27.124 path.go:setMappedLocations:555 ΓåÆ ~
[TRACE] 10:23:27.124 terminal.go:Flags() - 94ns
[DEBUG] 10:23:27.124 path.go:setMappedLocations:560 ΓåÆ mapped_locations: map[]
[TRACE] 10:23:27.124 terminal.go:GOOS() - 35ns
[TRACE] 10:23:27.124 terminal.go:PathSeparator() - 2.642┬╡s
[TRACE] 10:23:27.124 terminal.go:GOOS() - 54ns
[TRACE] 10:23:27.124 terminal.go:GOOS() - 26ns
[DEBUG] 10:23:27.124 segment.go:Execute:103 ΓåÆ segment: Python
[TRACE] 10:23:27.124 terminal.go:GOOS() - 34ns
[TRACE] 10:23:27.124 terminal.go:GOOS() - 40ns
[TRACE] 10:23:27.124 terminal.go:GOOS() - 24ns
[DEBUG] 10:23:27.124 python.go:Enabled:56 ΓåÆ environment
[TRACE] 10:23:27.124 terminal.go:PathSeparator() - 2.323┬╡s
[TRACE] 10:23:27.124 terminal.go:GOOS() - 39ns
[TRACE] 10:23:27.124 terminal.go:GOOS() - 33ns
[DEBUG] 10:23:27.124 language.go:Enabled:117 ΓåÆ extensions: [*.py *.ipynb pyproject.toml venv.bak]
[TRACE] 10:23:27.124 terminal.go:GOOS() - 31ns
[TRACE] 10:23:27.124 terminal.go:GOOS() - 48ns
[DEBUG] 10:23:27.124 language.go:Enabled:118 ΓåÆ folders: [.venv venv virtualenv venv-win pyenv-win]
[TRACE] 10:23:27.124 terminal.go:GOOS() - 40ns
[DEBUG] 10:23:27.124 language.go:Enabled:125 ΓåÆ home_enabled: false
[TRACE] 10:23:27.124 terminal.go:GOOS() - 33ns
[TRACE] 10:23:27.124 terminal.go:GOOS() - 28ns
[DEBUG] 10:23:27.124 python.go:loadContext:64 ΓåÆ fetch_virtual_env: true
[TRACE] 10:23:27.124 terminal.go:GOOS() - 38ns
[DEBUG] 10:23:27.124 path.go:makeFolderFormatMap:827 ΓåÆ NO DATA
[TRACE] 10:23:27.124 terminal.go:GOOS() - 27ns
[DEBUG] 10:23:27.124 path.go:setStyle:209 ΓåÆ full
[DEBUG] 10:23:27.124 path.go:colorizePath:728 ΓåÆ cycle: []
[DEBUG] 10:23:27.124 path.go:getFolderSeparator:263 ΓåÆ NO DATA
[DEBUG] 10:23:27.124 path.go:getFolderSeparator:265 ΓåÆ /
[DEBUG] 10:23:27.124 path.go:colorizePath:731 ΓåÆ cycle_folder_separator: false
[DEBUG] 10:23:27.124 path.go:colorizePath:732 ΓåÆ %s
[DEBUG] 10:23:27.124 path.go:colorizePath:734 ΓåÆ %s
[DEBUG] 10:23:27.124 path.go:colorizePath:735 ΓåÆ %s
[DEBUG] 10:23:27.124 path.go:colorizePath:736 ΓåÆ %s
[TRACE] 10:23:27.124 terminal.go:GOOS() - 70ns
[TRACE] 10:23:27.124 terminal.go:TemplateCache() - 330ns
[TRACE] 10:23:27.124 terminal.go:TemplateCache() - 109ns
[TRACE] 10:23:27.124 terminal.go:GOOS() - 55ns
[TRACE] 10:23:27.124 terminal.go:StackCount() - 82ns
[TRACE] 10:23:27.124 text.go:Render( {{ .Name }} ) - 222.276┬╡s
[TRACE] 10:23:27.124 text.go:Render(diamond) - 201ns
[TRACE] 10:23:27.124 terminal_unix.go:DirIsWritable(/home/heaths/.config/oh-my-posh) - 7.325┬╡s
[TRACE] 10:23:27.124 terminal.go:TemplateCache() - 35ns
[TRACE] 10:23:27.124 terminal.go:Flags() - 148ns
[TRACE] 10:23:27.125 terminal.go:GOOS() - 129ns
[DEBUG] 10:23:27.125 segment.go:Execute:103 ΓåÆ segment: Root
[TRACE] 10:23:27.125 terminal_unix.go:Root() - 436ns
[TRACE] 10:23:27.125 terminal.go:Flags() - 315.614┬╡s
[TRACE] 10:23:27.125 terminal.go:GOOS() - 79ns
[DEBUG] 10:23:27.125 segment.go:Execute:103 ΓåÆ segment: Status
[TRACE] 10:23:27.125 terminal.go:StatusCodes() - 87ns
[DEBUG] 10:23:27.125 status.go:formatStatus:45 ΓåÆ {{ .Code }}
[TRACE] 10:23:27.125 text.go:Render({{ .Code }}) - 43.483┬╡s
[DEBUG] 10:23:27.125 status.go:Enabled:37 ΓåÆ always_enabled: false
[DEBUG] 10:23:27.126 terminal.go:RunCommand:371 ΓåÆ NO DATA
[TRACE] 10:23:27.126 terminal.go:RunCommand(who am i) - 1.709386ms
[TRACE] 10:23:27.126 terminal.go:TemplateCache() - 207ns
[ERROR] 10:23:27.237 terminal.go:CommandPath:399 ΓåÆ exec: "python": executable file not found in $PATH
[TRACE] 10:23:27.237 terminal.go:CommandPath(python) - 112.638325ms
[DEBUG] 10:23:27.237 terminal.go:CommandPath:395 ΓåÆ /usr/bin/python3
[TRACE] 10:23:27.237 terminal.go:CommandPath(python3) - 52.048┬╡s
[DEBUG] 10:23:27.238 terminal.go:HasFilesInDir:243 ΓåÆ false
[TRACE] 10:23:27.238 terminal.go:HasFilesInDir(pyvenv.cfg) - 1.211953ms
[DEBUG] 10:23:27.238 terminal.go:HasFilesInDir:243 ΓåÆ false
[TRACE] 10:23:27.238 terminal.go:HasFilesInDir(pyvenv.cfg) - 44.655┬╡s
[DEBUG] 10:23:27.238 python.go:loadContext:77 ΓåÆ folder_name_fallback: true
[DEBUG] 10:23:27.238 python.go:loadContext:78 ΓåÆ default_venv_names: [.venv venv]
[DEBUG] 10:23:27.238 terminal.go:Getenv:161 ΓåÆ NO DATA
[TRACE] 10:23:27.238 terminal.go:Getenv(VIRTUAL_ENV) - 5.104┬╡s
[DEBUG] 10:23:27.238 terminal.go:Getenv:161 ΓåÆ NO DATA
[TRACE] 10:23:27.238 terminal.go:Getenv(CONDA_ENV_PATH) - 3.337┬╡s
[DEBUG] 10:23:27.238 terminal.go:Getenv:161 ΓåÆ NO DATA
[TRACE] 10:23:27.238 terminal.go:Getenv(CONDA_DEFAULT_ENV) - 2.401┬╡s
[TRACE] 10:23:27.238 text.go:Render(powerline) - 678ns
[TRACE] 10:23:27.238 text.go:Render(powerline) - 220ns
[TRACE] 10:23:27.238 text.go:Render(powerline) - 64ns
[TRACE] 10:23:27.238 terminal.go:TemplateCache() - 444ns
[TRACE] 10:23:27.238 text.go:Render( {{ if .SSHSession }}∩áù {{ end }}{{ .UserName }} ) - 203.065┬╡s
[TRACE] 10:23:27.238 text.go:Render(powerline) - 110ns
[TRACE] 10:23:27.239 text.go:Render(powerline) - 170ns
[TRACE] 10:23:27.239 text.go:Render( {{ path .Path .Location }} {{ repeat .StackCount "εé▒" }}) - 69.219┬╡s
[TRACE] 10:23:27.239 text.go:Render(powerline) - 133ns
[TRACE] 10:23:27.239 terminal.go:Shell() - 639ns
[TRACE] 10:23:27.239 terminal.go:Shell() - 28ns
[TRACE] 10:23:27.239 terminal.go:Flags() - 80ns
[TRACE] 10:23:27.239 terminal.go:GOOS() - 115ns
[DEBUG] 10:23:27.239 segment.go:Execute:103 ΓåÆ segment: Git
[DEBUG] 10:23:27.239 terminal_unix.go:IsWsl:35 ΓåÆ true
[TRACE] 10:23:27.239 terminal_unix.go:IsWsl() - 4.85┬╡s
[DEBUG] 10:23:27.239 terminal.go:FileContent:302 Γåô
    5.15.153.1-microsoft-standard-WSL2
    
[TRACE] 10:23:27.239 terminal.go:FileContent(/proc/sys/kernel/osrelease) - 51.866┬╡s
[TRACE] 10:23:27.239 terminal_unix.go:IsWsl2() - 69.395┬╡s
[DEBUG] 10:23:27.240 terminal.go:RunCommand:371 ΓåÆ //wsl.localhost/Ubuntu-22.04/home/heaths/.config/oh-my-posh
[TRACE] 10:23:27.240 terminal.go:RunCommand(wslpath -m /home/heaths/.config/oh-my-posh) - 944.214┬╡s
[TRACE] 10:23:27.240 terminal.go:GOOS() - 55ns
[DEBUG] 10:23:27.240 terminal.go:CommandPath:395 ΓåÆ /usr/bin/git
[TRACE] 10:23:27.240 terminal.go:CommandPath(git) - 40.764┬╡s
[TRACE] 10:23:27.240 terminal.go:HasCommand(git) - 43.388┬╡s
[DEBUG] 10:23:27.240 git.go:shouldDisplay:326 ΓåÆ fetch_bare_info: false
[DEBUG] 10:23:27.240 terminal.go:ResolveSymlink:287 ΓåÆ /home/heaths/.config/oh-my-posh
[TRACE] 10:23:27.240 terminal.go:ResolveSymlink(/home/heaths/.config/oh-my-posh) - 7.613┬╡s
[TRACE] 10:23:27.240 terminal.go:HasParentFilePath(.git) - 25.589┬╡s
[TRACE] 10:23:27.240 terminal.go:GOOS() - 30ns
[TRACE] 10:23:27.240 terminal.go:GOOS() - 21ns
[DEBUG] 10:23:27.240 git.go:Enabled:172 ΓåÆ fetch_user: false
[TRACE] 10:23:27.240 terminal.go:GOOS() - 23ns
[TRACE] 10:23:27.240 terminal.go:GOOS() - 20ns
[TRACE] 10:23:27.240 terminal.go:GOOS() - 20ns
[TRACE] 10:23:27.240 terminal.go:GOOS() - 20ns
[TRACE] 10:23:27.240 terminal.go:GOOS() - 20ns
[TRACE] 10:23:27.240 terminal.go:GOOS() - 20ns
[TRACE] 10:23:27.240 terminal.go:GOOS() - 22ns
[TRACE] 10:23:27.240 terminal.go:GOOS() - 20ns
[TRACE] 10:23:27.240 terminal.go:GOOS() - 21ns
[TRACE] 10:23:27.240 terminal.go:GOOS() - 20ns
[TRACE] 10:23:27.240 terminal.go:GOOS() - 23ns
[DEBUG] 10:23:27.240 git.go:Enabled:187 ΓåÆ cli
[DEBUG] 10:23:27.240 git.go:Enabled:192 ΓåÆ fetch_status: true
[DEBUG] 10:23:27.240 git.go:shouldIgnoreStatus:464 ΓåÆ ignore_status: []
[TRACE] 10:23:27.240 terminal.go:GOOS() - 21ns
[DEBUG] 10:23:27.240 git.go:setGitStatus:610 ΓåÆ status_formats: map[]
[DEBUG] 10:23:27.240 git.go:getSwitchMode:909 ΓåÆ untracked_modes: map[]
[DEBUG] 10:23:27.240 git.go:getSwitchMode:909 ΓåÆ ignore_submodules: map[]
[DEBUG] 10:23:27.241 terminal.go:RunCommand:371 Γåô
    # branch.oid 96ee23e3a1c28dc40938d48618f320756ce788c4
    # branch.head main
    # branch.upstream origin/main
    # branch.ab +0 -0
    1 .M N... 100755 100755 100755 c6462ce2ca3e96d863af2984c274db6177a5c110 c6462ce2ca3e96d863af2984c274db6177a5c110 theme.omp.yml
    ? theme.omp.yml.bak
[TRACE] 10:23:27.241 terminal.go:RunCommand(git -C /home/heaths/.config/oh-my-posh --no-optional-locks -c core.quotepath=false -c color.status=false status -unormal --branch --porcelain=2) - 1.506458ms
[DEBUG] 10:23:27.241 git.go:setGitHEADContext:669 ΓåÆ εéá 
[DEBUG] 10:23:27.241 scm.go:formatBranch:106 ΓåÆ mapped_branches: map[]
[DEBUG] 10:23:27.241 scm.go:formatBranch:122 ΓåÆ full_branch_path: true
[DEBUG] 10:23:27.241 scm.go:formatBranch:128 ΓåÆ branch_max_length: 0
[DEBUG] 10:23:27.241 terminal.go:HasFolder:272 ΓåÆ false
[TRACE] 10:23:27.241 terminal.go:HasFolder(/home/heaths/.config/oh-my-posh/.git/rebase-merge) - 5.705┬╡s
[DEBUG] 10:23:27.241 terminal.go:HasFolder:272 ΓåÆ false
[TRACE] 10:23:27.241 terminal.go:HasFolder(/home/heaths/.config/oh-my-posh/.git/rebase-apply) - 2.75┬╡s
[DEBUG] 10:23:27.241 git.go:setGitHEADContext:720 ΓåÆ ∩Éù
[DEBUG] 10:23:27.241 terminal.go:HasFilesInDir:243 ΓåÆ false
[TRACE] 10:23:27.241 terminal.go:HasFilesInDir(MERGE_MSG) - 39.657┬╡s
[DEBUG] 10:23:27.241 terminal.go:HasFilesInDir:243 ΓåÆ false
[TRACE] 10:23:27.241 terminal.go:HasFilesInDir(CHERRY_PICK_HEAD) - 4.322┬╡s
[DEBUG] 10:23:27.241 terminal.go:HasFilesInDir:243 ΓåÆ false
[TRACE] 10:23:27.241 terminal.go:HasFilesInDir(REVERT_HEAD) - 3.653┬╡s
[DEBUG] 10:23:27.241 terminal.go:HasFilesInDir:243 ΓåÆ false
[TRACE] 10:23:27.241 terminal.go:HasFilesInDir(sequencer/todo) - 3.54┬╡s
[DEBUG] 10:23:27.241 git.go:483 ΓåÆ Γëí
[DEBUG] 10:23:27.241 git.go:Enabled:205 ΓåÆ fetch_upstream_icon: false
[TRACE] 10:23:27.241 terminal.go:TemplateCache() - 176ns
[TRACE] 10:23:27.242 terminal.go:TemplateCache() - 70ns
[TRACE] 10:23:27.242 text.go:Render( {{ if .UpstreamURL }}{{ url .UpstreamIcon .UpstreamURL }}{{ end }}{{ .HEAD }}{{if .BranchStatus }} {{ .BranchStatus }}{{ end }}{{ if .Working.Changed }} ∩üä {{ .Working.String }}{{ end }}{{ if .Staging.Changed }} ∩üå {{ .Staging.String }}{{ end }} ) - 208.891┬╡s
[TRACE] 10:23:27.242 text.go:Render({{ if or (.Working.Changed) (.Staging.Changed) }}p:yellow{{ end }}) - 24.988┬╡s
[TRACE] 10:23:27.242 text.go:Render({{ if or (.Working.Changed) (.Staging.Changed) }}p:black{{ end }}) - 15.828┬╡s
[TRACE] 10:23:27.242 text.go:Render(diamond) - 159ns
[TRACE] 10:23:27.242 text.go:Render({{ if or (.Working.Changed) (.Staging.Changed) }}p:yellow{{ end }}) - 17.06┬╡s
[TRACE] 10:23:27.242 text.go:Render({{ if or (.Working.Changed) (.Staging.Changed) }}p:black{{ end }}) - 15.43┬╡s
[TRACE] 10:23:27.242 terminal.go:Shell() - 148ns
[TRACE] 10:23:27.242 terminal.go:Shell() - 23ns
[TRACE] 10:23:27.242 terminal.go:Flags() - 47ns
[TRACE] 10:23:27.242 terminal.go:Shell() - 25ns
[DEBUG] 10:23:27.242 terminal.go:Getenv:161 ΓåÆ NO DATA
[TRACE] 10:23:27.242 terminal.go:Getenv(LOCALAPPDATA) - 3.35┬╡s
[DEBUG] 10:23:27.242 terminal.go:Getenv:161 ΓåÆ NO DATA
[TRACE] 10:23:27.242 terminal.go:Getenv(OMP_CACHE_DIR) - 1.806┬╡s
[DEBUG] 10:23:27.242 terminal.go:Getenv:161 ΓåÆ NO DATA
[TRACE] 10:23:27.242 terminal.go:Getenv(XDG_CACHE_HOME) - 1.897┬╡s
[TRACE] 10:23:27.242 terminal.go:CachePath() - 21.117┬╡s
[TRACE] 10:23:27.242 terminal.go:Flags() - 44ns
```

--- a/src/cli/disable.go
+++ b/src/cli/disable.go
@@ -2,13 +2,14 @@ package cli
 
 import (
 	"fmt"
+	"strings"
 
 	"github.com/spf13/cobra"
 )
 
 // getCmd represents the get command
 var disableCmd = &cobra.Command{
-	Use:       fmt.Sprintf(toggleUse, "disable"),
+	Use:       fmt.Sprintf(toggleUse, "disable", strings.Join(toggleArgs, "|")),
 	Short:     "Disable a feature",
 	Long:      fmt.Sprintf(toggleLong, "Disable", "disable"),
 	ValidArgs: toggleArgs,

--- a/src/cli/enable.go
+++ b/src/cli/enable.go
@@ -2,6 +2,7 @@ package cli
 
 import (
 	"fmt"
+	"strings"
 
 	"github.com/jandedobbeleer/oh-my-posh/src/cache"
 	"github.com/jandedobbeleer/oh-my-posh/src/config"
@@ -11,12 +12,12 @@ import (
 )
 
 var (
-	toggleUse  = "%s [notice]"
+	toggleUse  = "%s [%s]"
 	toggleLong = `%s a feature
 
 This command is used to %s one of the following features:
 
-- notice`
+- upgradenotice`
 	toggleArgs = []string{
 		config.UPGRADENOTICE,
 		config.AUTOUPGRADE,
@@ -25,7 +26,7 @@ This command is used to %s one of the following features:
 
 // getCmd represents the get command
 var enableCmd = &cobra.Command{
-	Use:       fmt.Sprintf(toggleUse, "enable"),
+	Use:       fmt.Sprintf(toggleUse, "enable", strings.Join(toggleArgs, "|")),
 	Short:     "Enable a feature",
 	Long:      fmt.Sprintf(toggleLong, "Enable", "enable"),
 	ValidArgs: toggleArgs,
@@ -54,24 +55,15 @@ func toggleFeature(cmd *cobra.Command, feature string, enable bool) {
 	env.Init()
 	defer env.Close()
 
-	var key string
-
-	switch feature {
-	case "notice":
-		key = config.UPGRADENOTICE
-	case "autoupgrade":
-		key = config.AUTOUPGRADE
-	}
-
-	if len(key) == 0 {
+	if len(feature) == 0 {
 		_ = cmd.Help()
 		return
 	}
 
 	if enable {
-		env.Cache().Set(key, "true", cache.INFINITE)
+		env.Cache().Set(feature, "true", cache.INFINITE)
 		return
 	}
 
-	env.Cache().Delete(key)
+	env.Cache().Delete(feature)
 }

--- a/src/config/config.go
+++ b/src/config/config.go
@@ -14,8 +14,8 @@ const (
 	YAML string = "yaml"
 	TOML string = "toml"
 
-	AUTOUPGRADE   = "autoupgrade"
-	UPGRADENOTICE = "upgradenotice"
+	AUTOUPGRADE   = "upgrade"
+	UPGRADENOTICE = "notice"
 
 	Version = 3
 )

--- a/src/config/segment.go
+++ b/src/config/segment.go
@@ -46,10 +46,10 @@ type Segment struct {
 	LeadingDiamond         string         `json:"leading_diamond,omitempty" toml:"leading_diamond,omitempty"`
 	TrailingDiamond        string         `json:"trailing_diamond,omitempty" toml:"trailing_diamond,omitempty"`
 	Template               string         `json:"template,omitempty" toml:"template,omitempty"`
-	Foreground             color.Ansi     `json:"foreground" toml:"foreground"`
+	Foreground             color.Ansi     `json:"foreground,omitempty" toml:"foreground,omitempty"`
 	TemplatesLogic         template.Logic `json:"templates_logic,omitempty" toml:"templates_logic,omitempty"`
 	PowerlineSymbol        string         `json:"powerline_symbol,omitempty" toml:"powerline_symbol,omitempty"`
-	Background             color.Ansi     `json:"background" toml:"background"`
+	Background             color.Ansi     `json:"background,omitempty" toml:"background,omitempty"`
 	Filler                 string         `json:"filler,omitempty" toml:"filler,omitempty"`
 	Type                   SegmentType    `json:"type,omitempty" toml:"type,omitempty"`
 	Alias                  string         `json:"alias,omitempty" toml:"alias,omitempty"`

--- a/src/prompt/engine.go
+++ b/src/prompt/engine.go
@@ -166,6 +166,13 @@ func (e *Engine) getTitleTemplateText() string {
 }
 
 func (e *Engine) renderBlock(block *config.Block, cancelNewline bool) bool {
+	text, length := e.writeBlockSegments(block)
+
+	// do not print anything when we don't have any text
+	if length == 0 {
+		return false
+	}
+
 	defer e.applyPowerShellBleedPatch()
 
 	// do not print a newline to avoid a leading space
@@ -173,13 +180,6 @@ func (e *Engine) renderBlock(block *config.Block, cancelNewline bool) bool {
 	// the shell
 	if block.Newline && !cancelNewline {
 		e.writeNewline()
-	}
-
-	text, length := e.writeBlockSegments(block)
-
-	// do not print anything when we don't have any text
-	if length == 0 {
-		return false
 	}
 
 	switch block.Type {

--- a/website/docs/installation/upgrade.mdx
+++ b/website/docs/installation/upgrade.mdx
@@ -29,7 +29,7 @@ oh-my-posh upgrade
 <TabItem value="cli">
 
 ```powershell
-oh-my-posh enable autoupgrade
+oh-my-posh enable upgrade
 ```
 
 </TabItem>

